src/csharp/Grpc.Tools/ProtoCompile.cs

```diff
@@ -211,7 +211,7 @@ public class ProtoCompile : ToolTask
             new ErrorListFilter
             {
                 Pattern = new Regex(
-                    pattern: "(?'FILENAME'.+): ?(?'TEXT'.*)",
+                    pattern: "^(?'FILENAME'.+?): ?(?'TEXT'.*)",
```

Comment:
While we're on the topic of valid file names, there's a problem here. `:` is allowed as a filename character on some OS's which can throw off the parsing here. One option is to make the space after the colon mandatory which might help.

Reply:
I think we can't really do much better here. This rule is only a heuristic and if someone really wants to have .proto file called e.g. `"foo bar: the final version.proto"` we can't stop him. I think it's acceptable if VS error will be parsed incorrectly in such case.  In my experience the colon .proto file names would be less common that parenthesis (some editors create copies of files in form `foobar (copy 1).proto` so it's good to handle that case)

the nongreedy version of filename group is good here because at least we won't spend too much time matching the file name if the error message is long.